clients/drcachesim/tools/filter/record_filter.h

```diff
@@ -66,18 +66,25 @@ class record_filter_t : public record_analysis_tool_t {
          * any entry. The returned pointer is passed to all invocations of
          * parallel_shard_filter() and parallel_shard_exit().
          * This routine can be used to initialize state for each shard.
+         * \p partial_trace_filter denotes whether the trace will be filtered
+         * only partially, e.g. due to stop_timestamp.
          */
         virtual void *
-        parallel_shard_init(memtrace_stream_t *shard_stream) = 0;
+        parallel_shard_init(memtrace_stream_t *shard_stream,
+                            bool partial_trace_filter) = 0;
```

Comment:
Should this be a flag so we can add more later -- but see below: I'm not sure I understand the use case for this.

Reply:
Not sure what you mean by flag here?

Reply:
A bitmask value inside an integer var so we could add more w/o adding more bool params.  Are we sure we won't want more qualifiers later?

Reply:
Or even use default args.

Reply:
I don't feel strongly either/any way.

Reply:
I'll leave it to later then, since currently I don't see a requirement.